trustpinning/certs.go

```diff
@@ -175,6 +175,17 @@ func ValidateRoot(prevRoot *data.SignedRoot, root *data.Signed, gun string, trus
 	return data.RootFromSigned(root)
 }
 
+// MatchCNToGun checks that the common name in a cert is valid for the given gun.
+// This allows wildcards as suffixes, e.g. `namespace/*`
+func MatchCNToGun(commonName, gun string) bool {
+	const wildcard = "*"
+	if strings.HasSuffix(commonName, wildcard) {
+		prefix := commonName[:len(commonName)-len(wildcard)]
```

Comment:
Also, could we add a log debug to display which prefix we're comparing against?

My concern is if folks use something like `prefix**` and expect matching against `prefix` instead of `prefix*`